lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1173,6 +1202,28 @@ class ConvertAtenMatmulBaseOp : public OpConversionPattern<AtenOpT> {
       return rewriter.notifyMatchFailure(op,
                                          "Matmul: input datatypes mismatched");
 
+    // Step: check if the inputs have been casted from a supported input type to
+    // an accumulator type and insert casts back to the original type if true
+    RankedTensorType lhsPreCastedType =
+        getCastedInputTypeForMatmul(lhs, rewriter);
+    RankedTensorType rhsPreCastedType =
+        getCastedInputTypeForMatmul(rhs, rewriter);
+    if (lhsPreCastedType && (lhsPreCastedType.getElementType() ==
+                             rhsPreCastedType.getElementType())) {
```

Comment:
`rhsPreCastedType` could also be "null" RankedTensorType

Reply:
done